pkg/semtech-loramac/Makefile.loramac_mac

```diff
@@ -0,0 +1,5 @@
+MODULE := semtech_loramac_mac
+
+CFLAGS += -Wno-missing-field-initializers -Wno-unused-parameter -Wno-sign-compare
```

Comment:
I just moved this CFLAGS rules here and [here](https://github.com/RIOT-OS/RIOT/pull/8264/files#diff-4f32d7da4e468a1350aea6570389f140R3) so that it's not applied globally to the package module